performance_test/design/performance_test-design.md

```diff
@@ -26,9 +26,11 @@ The tool has a fully documented command line interface which can be accessed by
 To run more extensive tests, the following script can be used and adapted:
 `helper_scripts/run_experiment.py`
 
-The logfiles can be visualized using the following python script:
-`helper_scripts/performance_test_file_reader.py`
-Note that `pandas` is required to run this script.
+The logfiles can be visualized using the apex_performance_plotter module:
+```
+pip3 install performance_test/helper_scripts/apex_performance_plotter
+perfplot NAME_OF_LOG_FILE
```

Comment:
@pbaughman After executing the instructions to install the performance plotter I am getting the following error:
```

divya.aggarwal@ade:~/perf_test_ws$ perfplot log_Array16k_23-09-2019_17-17-55
bash: perfplot: command not found
```

Reply:
@daggarwa Can you please give me more information?  It's not possible to figure out what went wrong with just

```
bash: perfplot: command not found
```

Obviously something failed when you tried to install it - was there an error message when you pip3 installed apex_performance_plogger

Reply:
@pbaughman Here is the log for the installation :

```
divya.aggarwal@ade:~/perf_test_ws/src/performance_test (master *$% u=)$ pip3 install performance_test/helper_scripts/apex_performance_plotter
Processing ./performance_test/helper_scripts/apex_performance_plotter
Collecting Click (from apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl
Collecting Jinja2 (from apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/1d/e7/fd8b501e7a6dfe492a433deb7b9d833d39ca74916fa8bc63dd1a4947a671/Jinja2-2.10.1-py2.py3-none-any.whl
Collecting pandas (from apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/73/9b/52e228545d14f14bb2a1622e225f38463c8726645165e1cb7dde95bfe6d4/pandas-0.25.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting MarkupSafe>=0.23 (from Jinja2->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/b2/5f/23e0023be6bb885d00ffbefad2942bc51a620328ee910f64abe5a8d18dd1/MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting python-dateutil>=2.6.1 (from pandas->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/41/17/c62faccbfbd163c7f57f3844689e3a78bae1f403648a6afb1d0866d87fbb/python_dateutil-2.8.0-py2.py3-none-any.whl
Collecting numpy>=1.13.3 (from pandas->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/e5/e6/c3fdc53aed9fa19d6ff3abf97dfad768ae3afce1b7431f7500000816bda5/numpy-1.17.2-cp36-cp36m-manylinux1_x86_64.whl
Collecting pytz>=2017.2 (from pandas->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/87/76/46d697698a143e05f77bec5a526bf4e56a0be61d63425b68f4ba553b51f2/pytz-2019.2-py2.py3-none-any.whl
Collecting six>=1.5 (from python-dateutil>=2.6.1->pandas->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Installing collected packages: Click, MarkupSafe, Jinja2, six, python-dateutil, numpy, pytz, pandas, apex-performance-plotter
  Running setup.py install for apex-performance-plotter ... done
Successfully installed Click-7.0 Jinja2-2.10.1 MarkupSafe-1.1.1 apex-performance-plotter-0.1.0 numpy-1.17.2 pandas-0.25.1 python-dateutil-2.8.0 pytz-2019.2 six-1.12.0

```
I am not seeing any errors in the installation here.

Reply:
Sorry, I'm not sure what to tell you then.

Reply:
Here's mine:

```
pete.baughman@ade:~/performance_test (remove_old_plotter $%)$ pip3 install performance_test/helper_scripts/apex_performance_plotter/
Processing ./performance_test/helper_scripts/apex_performance_plotter
Collecting Click (from apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl
Collecting Jinja2 (from apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/1d/e7/fd8b501e7a6dfe492a433deb7b9d833d39ca74916fa8bc63dd1a4947a671/Jinja2-2.10.1-py2.py3-none-any.whl
Collecting pandas (from apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/73/9b/52e228545d14f14bb2a1622e225f38463c8726645165e1cb7dde95bfe6d4/pandas-0.25.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting MarkupSafe>=0.23 (from Jinja2->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/b2/5f/23e0023be6bb885d00ffbefad2942bc51a620328ee910f64abe5a8d18dd1/MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting numpy>=1.13.3 (from pandas->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/e5/e6/c3fdc53aed9fa19d6ff3abf97dfad768ae3afce1b7431f7500000816bda5/numpy-1.17.2-cp36-cp36m-manylinux1_x86_64.whl
Collecting pytz>=2017.2 (from pandas->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/87/76/46d697698a143e05f77bec5a526bf4e56a0be61d63425b68f4ba553b51f2/pytz-2019.2-py2.py3-none-any.whl
Collecting python-dateutil>=2.6.1 (from pandas->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/41/17/c62faccbfbd163c7f57f3844689e3a78bae1f403648a6afb1d0866d87fbb/python_dateutil-2.8.0-py2.py3-none-any.whl
Collecting six>=1.5 (from python-dateutil>=2.6.1->pandas->apex-performance-plotter==0.1.0)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Installing collected packages: Click, MarkupSafe, Jinja2, numpy, pytz, six, python-dateutil, pandas, apex-performance-plotter
  Running setup.py install for apex-performance-plotter ... done
Successfully installed Click-7.0 Jinja2-2.10.1 MarkupSafe-1.1.1 apex-performance-plotter-0.1.0 numpy-1.17.2 pandas-0.25.1 python-dateutil-2.8.0 pytz-2019.2 six-1.12.0
pete.baughman@ade:~/performance_test (remove_old_plotter $%)$ perfplot
Usage: perfplot [OPTIONS] FILENAMES...
Try "perfplot --help" for help.

Error: Missing argument "FILENAMES...".
```

```

Reply:
@pbaughman As we discussed offline , there was an issue with my local environment where the PATH env variable was not including `./local/bin` which contained the installed `perfplot` binary and hence it was not found. I have resolved the issue by doing a fresh setup of my local environment and it works fine now.